pandas/tests/indexes/test_old_base.py

```diff
@@ -228,6 +233,8 @@ def test_logical_compat(self, simple_index):
                     r"'IntervalArray' with dtype interval\[.*\] does "
                     "not support reduction '(any|all)'"
                 )
+            if isinstance(idx, PeriodIndex):
```

Comment:
I think you can combine this branch with the IntervalArray branch directly preceding it and reduce the msg check to `does not support reduction '(any|all)'`

It is generally a good idea to try and reduce the amount of branching in tests. Having a bunch of if...else statements makes it unnecessarily harder to manage consistent behavior expectations